packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -107,9 +112,49 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
       this.startupTimeout
     );
 
+    await Promise.race([this.waitContainerNotExited(container, waitingFinished), healthCheckPromise])
+      .finally(() => waitingFinished.value = true)
+
     log.debug(`HTTP wait strategy complete`, { containerId: container.id });
   }
 
+  private async waitContainerNotExited(container: Dockerode.Container, waitingFinished: {value: boolean}) {
+    const exitStatus = 'exited';
+    const status = await new IntervalRetry<string, null>(500).retryUntil(
+      async () => (await container.inspect()).State.Status,
+      (status) => waitingFinished.value || status === exitStatus,
+      () => null,
+      this.startupTimeout + 500 // delay for timeout after healthCheck
+    );
+
+    if (status !== exitStatus) {
+      return
+    }
+
+    const tail = 50;
+    const lastLogs: string[] = [];
+    const client = await getContainerRuntimeClient();
+    let message: string;
+
+    try {
+      const stream = await client.container.logs(container, {tail});
```

Comment:
Get logs through client because if we get logs directly from Dockerode.Container we will have [bad symbols](https://github.com/apocas/dockerode/issues/469) (frame headers).